### ISSUE TITLE
Can't kick certain player(s) in casual that haven't picked class

### ISSUE THREAD

**Issue by jh34ghu43gu** [ValveSoftware/Source-1-Games#5187]
Don't know if the fact they haven't picked a class is related, I'm pretty sure I've kicked people before they picked a class previously.

This guy has been hanging around in dustbowl lobbies for the past few days taking up a player slot and unable to be kicked by anyone.
https://www.youtube.com/watch?v=6bVGDdu41es

**Comment by jh34ghu43gu** [comment]
I should add I'm not sure if #5092 is related since that specifically deals with spectator team players (since it mentions being able to see them in the kick menu which the person in the video I am able to see.)

**Comment by NULLYUKI** [comment]
> I should add I'm not sure if #5092 is related since that specifically deals with spectator team players (since it mentions being able to see them in the kick menu which the person in the video I am able to see.)

Personally I would say, that your issue is a separate issue compared to #5092. As yours is occurring when a player does not pick a class and the other issue occurs when a player is a spectator.

But on the other hand it could be also a issue that could be combined, with the information that under certain conditions, a player cannot be kicked.

**Comment by Ashetf2** [comment]
This issue is relevant now. Please fix.

**Comment by NotHyper-474** [comment]
According to [this post](https://www.reddit.com/r/tf2/s/z2io9XUUB5) on r/tf2, (some) bots are now abusing this to avoid being kicked

**Comment by PocketSpade** [comment]
Yeah we need to be able to kick spectator/unassigned players, currently even more of an issue due to bot hosters exploiting this to fill casual matches with inactive players
